### ISSUE TITLE
Block custom message when block user but alert message received is there anyway to solve this problem.

### ISSUE THREAD

**Issue by githubmss** [QuickBlox/quickblox-android-sdk#463]
I have a particular case in my project sir , your block and unblock functionality is working fine in case of chat , But here is where i am getting problem , In my Application user can even send alert messages to another user even if that user is blocked by other user. 

**Explanation of Flow :**- 

Suppose i have 2 users , User A send an alert message to User B which will create chat dialog between user A and B . Now suppose for any reason User A block the user B , in that case both can not send message in chat this is working absolutely fine using QBPrivacylist , But my case is , they can not send messages in chat but can send Alert messages to each other. So is there any way where i can make condition that while sending alert messages i can unblock both of them and once alert message reached i will block both of them again .Please suggest a way.  

**IS there any way USER B can get list of peoples who blocked him/her ?**

public void onClickBlock() {
        qbPrivacyList.setName(String.valueOf(userId) + String.valueOf(recipientId));
        qbPrivacyList.setActiveList(true);
        ArrayList<QBPrivacyListItem> items = new ArrayList<QBPrivacyListItem>();
        QBPrivacyListItem item = new QBPrivacyListItem();
        item.setAllow(false);
        item.setType(QBPrivacyListItem.Type.USER_ID);
        item.setValueForType(String.valueOf(recipientId));
        item.setMutualBlock(false);
        items.add(item);
        qbPrivacyList.setItems(items);
        qbPrivacyList.setDefaultList(true);
        try {
            privacyListsManager.setPrivacyListAsDefault(String.valueOf(userId) + String.valueOf(recipientId));
            privacyListsManager.setPrivacyListAsActive(String.valueOf(userId) + String.valueOf(recipientId));
            privacyListsManager.setPrivacyList(qbPrivacyList);
        } catch (SmackException.NotConnectedException e) {
            e.printStackTrace();
        } catch (XMPPException.XMPPErrorException e) {
            e.printStackTrace();
        } catch (SmackException.NoResponseException e) {
            e.printStackTrace();
        }
    }

//Block and Unblock working fine. I just want to send a particular messages even in case of blocked users any solution please help me...

**Comment by RomanPronin** [comment]
Hi, @githubmss. I can offer you [send pushNotification](https://quickblox.com/developers/SimpleSample-messages_users-android#Send_Push_Notification_to_particular_users_.28through_their_IDs.29) to blocked user.

**Comment by kishorejethava** [comment]
Hello @RomanPronin 
Problem : 
User A can send Push Notification to User B using **Event** even User B is blocked 

Query : How to prevent send push notification in the case of user is blocked?

**Comment by RomanPronin** [comment]
Hello @kishorejethava, you can always implement your own logic on sender side. For example, check if user is blocked  and do not send push to him. Something like this:
```       
boolean isUserBlocked(String userID) {
        QBPrivacyList qbPrivacyList = chatService.getPrivacyListsManager().getPrivacyList("LIST_NAME");
        for (QBPrivacyListItem listItem : qbPrivacyList.getItems()) {
            if (listItem.getValueForType().contains(userID)) {
                return true;
            }
        }
        return false;
    }

**Comment by maughray** [comment]
> Hello @kishorejethava, you can always implement your own logic on sender side. For example, check if user is blocked and do not send push to him. Something like this:
> 
> ```
> boolean isUserBlocked(String userID) {
>         QBPrivacyList qbPrivacyList = chatService.getPrivacyListsManager().getPrivacyList("LIST_NAME");
>         for (QBPrivacyListItem listItem : qbPrivacyList.getItems()) {
>             if (listItem.getValueForType().contains(userID)) {
>                 return true;
>             }
>         }
>         return false;
>     }
> ```

I had the same problem. In this way you verify if user is in your privacy list, but the question is why notification are not blocked if partner has blocked you? You can't verify his privacy list.

**Comment by QB-nikolay-zolotarov** [comment]
Hello QuickBlox customer,

This is Nikolay from QuickBlox support.

The issue was closed as it is outdated.

Please update the SDK to the latest version:
[https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11](https://github.com/QuickBlox/quickblox-android-sdk-releases/releases/tag/3.9.11)

Additionally, please check our new samples:
[https://docs.quickblox.com/docs/code-samples#chat-samples](https://docs.quickblox.com/docs/code-samples#chat-samples)

If it is still relevant after reviewing the updated information, feel free to open a new issue.

Have a nice day.
